lib/fizzy/execute.cpp

```diff
@@ -478,14 +479,14 @@ void branch(const Code& code, OperandStack& stack, const uint8_t*& pc, const uin
 }
 
 template <class F>
-inline bool invoke_function(
-    const FuncType& func_type, const F& func, Instance& instance, OperandStack& stack, int depth)
+inline bool invoke_function(const FuncType& func_type, const F& func, Instance& instance,
```

Comment:
Probably it'll be better to have the `ThreadContext` change first applied and include `gas` within that. That would also allow for querying the remaining gas after execution has finished.